app/master/build_scheduler_pool.py

```diff
@@ -20,12 +21,16 @@ def get(self, build):
         :type build: Build
         :rtype: BuildScheduler
         """
+        if isinstance(build, Build):
```

Comment:
I think changes to this class might not be necessary if my suggestions in my previous comments are doable. If they are necessary though, my personal preference would be to add a `build_id` parameter to this method, then the validation logic here would check that exactly one of `build` or `build_id` is specified.

Reply:
Yes. I forgot to point out this. Its a hack which we used and we can remove some of the changes. I will update it and get back to you.
Also I tested the POST on "cancel" endpoint and it works.